contributors/design-proposals/federated-statefulsets.md

```diff
@@ -0,0 +1,249 @@
+
+## Scale up/Scale down
+
+When a federated statefulset is scaled up or down, the distribution of replicas will be recalculated and the replica numbers in each cluster will be updated independently.
+
+The behaviour of the scaling on being targeted using a **federated hpa** will be discussed in a separate design.
+
+## Handling additional LB services
+
+As listed in the approaches, the current option of exposing the statefulset pods across clusters is to assign a LB service to each replica pod.
+If the federated service controller is used (by creating the service in federation) to create these services, then as per the current behaviour the service will be created in each cluster. 
+This leads to exactly one unused service per statefulset per cluster.
+There are two alternatives to circumvent this problem:
+
+1 - Cluster affinity is introduced at least for services such that the controller creates the service only in needed cluster. (The need of cluster affinity and anti-affinity in general is discussed [elsewhere](https://github.com/kubernetes/kubernetes/issues/41442) also)
+
+2 - Federated statefulset controller handles the creation of cluster local services, rather than the federated service controller. It also will need to handle the dns records for each LB service into the federation (or cloud provider specific) public dns server.
+
+We propose using alternative 1 listed here, as it fits broader scheme of things and is more consistent with the user expectation of being able to query all needed resources from the federation control plane and is less confusing to use at the same time.
+
+# Conclusion
+
+Alternative 1 can solve only 1 of the listed use cases (scenario 1), and does not remain consistent with the existing federated controllers.
+
+Alternative 2 can solve both the use cases, is consistent with the existing federated controllers and makes more sense in the scheme of things.
+
+# Limitations (to be discussed)
+
+_1 – In in case of a particular replica or the cluster containing particular replicas die, what should be the behavior._
+
+It can be resurrected with the same identity in the original cluster only
+
+_2 – Is there a migration scenario? If there is, in case a replica needs migration, how do we handle this?_
+
+elaborated by @quinton-hoole
+
+Strictly speaking, migration between clusters in the same zone is quite feasible.
+
+I think we&#39;re left with a few options:
+
+(1) only allow the use of node-local persistent (bad),
+
+(2) disallow migration of replicas between clusters if they use non-node-local volumes (simple, but perhaps overly restrictive)
```

Comment:
This is the preferred method. Don't attempt to move storage across clusters. Most distributed databases have their own notion of identity, consistency and replication. This is an app specific concern, don't overcomplicate k8s by trying to manage replication for them.